### PR TITLE
Whitelistings multipackages

### PR DESCRIPTION
**Pull request by mgerstner**

### CODE REVIEW

**Comment by mgerstner** [comment]
I found a small coding style issue just now, hence the force-push.

**Review by marxin — CHANGES_REQUESTED** [review]
Please come up with a test-case for it.

File: rpmlint/checks/FileDigestCheck.py [hunk 1]
```diff
@@ -22,30 +22,62 @@ def __init__(self, config, output):
...
+        elif package and packages:
+            raise KeyError("FileDigestCheck: encountered both 'package' and 'packages' keys in FileDigestGroup entry")
+
+        if package and type(package) != str:
+            raise KeyError("FileDigestCheck: 'package' key contains non-string value")
+        elif packages and type(packages) != list:
+            raise KeyError("FileDigestCheck: 'packages' key contains non-list value")
+
+    def _matches_pkg(self, digest_group, pkg):
+        if pkg.name == digest_group.get('package', ""):
```
[the remark this answers is not in the document]
Comment (mgerstner) [reply rpmlint/checks/FileDigestCheck.py]: I don't think it got reported, there have been some existing lines already with different quoting. Changed this now.

File: configs/openSUSE/cron-whitelist.toml [hunk 2]
```diff
@@ -6,8 +6,8 @@ bug = "bsc#1188704"
 digests = [
     {
         path = "/etc/cron.hourly/rpmlint_cron",
-	algorithm = "sha256",
-	hash = "3cf3c05e5c37b0ee7b793f9c5c9b57b74071b02137a7e66a10a52d91e0ddbc51"
+        algorithm = "sha256",
```
[the remark this answers is not in the document]
Comment (mgerstner) [reply configs/openSUSE/cron-whitelist.toml]: I did so.

File: test/test_file_digest.py [hunk 3]
```diff
@@ -206,6 +206,22 @@ def test_combination_nodigests_and_digests():
         assert error == 'testpkg: E: somerestriction-file-digest-mismatch /related/and/also/sensitive expected sha1:ab5ec199027247773d2d617895f49179d7b3186e, has:a6abec9ea1e13ca93d1c704758bd52f62ef16433'
 
 
+def test_multiple_packages():
+
```
Comment (marxin) [inline test/test_file_digest.py:210]: Please remove the empty line.

**Review by marxin — CHANGES_REQUESTED** [review]
Please fix the failing test and I'm fine with the suggested changes.

File: rpmlint/checks/FileDigestCheck.py [hunk 4]
```diff
@@ -22,30 +22,62 @@ def __init__(self, config, output):
...
+        if dg_type not in self.digest_configurations:
+            raise KeyError(f'FileDigestGroup type "{dg_type}" is not '
+                           f'supported, known values: {list(self.digest_configurations)}')
+        # verify digest algorithm
+        for digest in digest_group['digests']:
+            algorithm = digest['algorithm']
+            if algorithm == 'skip':
+                pass
+            else:
+                hashlib.new(algorithm)
```
Comment (jsegitz) [inline rpmlint/checks/FileDigestCheck.py:59]: add some error handling here for misspelled algorithms

**Review by jsegitz — CHANGES_REQUESTED** [review]

File: rpmlint/checks/FileDigestCheck.py (hunk 4 above)
Comment (marxin) [reply rpmlint/checks/FileDigestCheck.py:59]: Well, I think the current Exception is reasonable: `ValueError: unsupported hash type sha256xx`.

Comment (mgerstner) [reply rpmlint/checks/FileDigestCheck.py:59]: Well that *is* the error handling, just not that obvious. If the spelling is wrong then `hashlib.new` will raise.

File: test/test_file_digest.py (hunk 3 above)
Comment (mgerstner) [reply test/test_file_digest.py]: done

**Comment by mgerstner** [comment]
I can't see why the CI tests are failing, the new unit test is working fine for me on a Tumbleweed VM.

Otherwise all change requests should be accomodated by now.

File: test/configs/digests.config [hunk 5]
```diff
@@ -48,3 +48,15 @@ digests = [
...
 ]
 
+[[FileDigestGroup]]
+packages = ["otherpkg", "testpkg2"]
+type = "somerestriction"
+note = "tests a package that has matching digests, using multiple package names"
+bug = "bsc#1234"
+digests = [
+    {
+        path = "/resetricted/1/afile",
```
Comment (marxin) [inline test/configs/digests.config:58]: It's because of this typo: `resetricted`, should be `restricted`.

Comment (mgerstner) [reply test/configs/digests.config]: Thanks for the hint. I must have been making some major mistake when running the tests. It should be correct now.

**Pull request merged**